main-app/app/scripts/game/services/bingo-ticket.js

```diff
@@ -1,41 +1,52 @@
-(function () {
+(function() {
     'use strict';
     angular.module('Tombola.Games.Bingo90.Core')
         .service('BingoTicket', [function() {
-            var me = this;
-            var tickets = [];
+            var me = this,
+                tickets = [],
+                lines = [];
 
             me.balls = [];
+            me.markedNumbers = [];
 
-            me.pushArray = function (card) {
-                var cardss = "054963758028345266770611596982054963758028345266770611596982";
-                generateTheTicket(card);
-            };
-
-            var generateTheTicket = function (ticket) {
-                var startOfString = 0;
-                var endOfString = 30;
-                for(startOfString; startOfString < ticket.length; startOfString += 30){
-                    tickets.push(ticket.slice(startOfString, endOfString));
-                    endOfString += 30;
-                }
-
-                for(ticket in tickets){
-                    splitUpTheTicket(tickets[ticket]);
+            me.pushArray = function(card) {
+                var cardss = "054963758028345266770611596982054963758028345266770611596982",
+                    i;
+                for(i = 0; i < 27; i++){
+                    me.markedNumbers[i] = "bingoCell";
                 }
+                generateTheTicket(card);
             };
 
-            var splitUpTheTicket = function (ticket) {
-                var startOfString = 0;
-                var endOfString = 2;
-                for(startOfString; startOfString < ticket.length; startOfString += 2){
-                    me.balls.push(ticket.slice(startOfString, endOfString));
-                    endOfString += 2;
-                }
-                console.log(me.balls);
-            };
-
-
+            var generateTheTicket = function(ticket) {
```

Comment:
Not going to continue to reviewing this file. 
Need to think more in  terms of objects:
1. A card has a an array of 4 lines. If all the lines have 0 to go, they you've won a house prize.
2. A line is made up of an array of 9 squares. You can calculate how many to go by adding up which squares are "to go".
3.  A square has a value or is null. It is "to go" if it is not null and the number hasn't been called.